### ISSUE TITLE
integration of Authentication Service

### ISSUE THREAD

**Issue by SimOneGollo** [RIP-Comm/AMAlanche#19]
It must be added an AuthenticationService with following functions:

- [ ] Login -> Return authentication token
- [ ] SingUp -> Create user and return authentication token

**Comment by matteasu** [comment]
Do we want to provide support for account creation trough 3rd party providers such as twitch or google?

**Comment by datge** [comment]
I think twitch and google signup/login are even more important then standard one being that most of the user are gonna came from twitch at this phase.

**Comment by SimOneGollo** [comment]
Yes, I think that we can add Google and/or Twitch integration, but i don't exclude our standard login/singup. After all the registration need only 3 value (username, email, password).

**Comment by datge** [comment]
right now i've just implmented standard sign-in,sign-up. task for google & twitch are still open.
